libs/metagen/src/mdk_rust/stubs.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright Metatype OÜ, licensed under the Elastic License 2.0.
+// SPDX-License-Identifier: Elastic-2.0
+
+use super::utils::normalize_type_title;
+use crate::interlude::*;
+use crate::mdk::*;
+use crate::utils::*;
+use std::fmt::Write;
+
+pub struct GenStubOptions {}
+
+pub fn gen_stub(
+    fun: &StubbedFunction,
+    dest: &mut GenDestBuf,
+    type_names: &HashMap<u32, Arc<str>>,
+    _opts: &GenStubOptions,
+) -> anyhow::Result<Arc<str>> {
+    let TypeNode::Function { base, data } = &fun.node else {
+        unreachable!()
+    };
+    let inp_ty = type_names
+        .get(&data.input)
+        .context("input type for function not found")?;
+    let out_ty = type_names
+        .get(&data.output)
+        .context("output type for function not found")?;
+    let trait_name: Arc<str> = normalize_type_title(&base.title).into();
+    writeln!(
+        &mut dest.buf,
+        r#"pub trait {trait_name} {{
+    fn handle(input: {inp_ty}, cx: Ctx) -> anyhow::Result<{out_ty}>;
+}}"#
+    )?;
+    Ok(trait_name)
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::{mdk_rust::*, tests::default_type_node_base};
+    use common::typegraph::*;
+
+    #[test]
+    fn stub_test() -> anyhow::Result<()> {
+        let tg_name = "my_tg".to_string();
+        let tg = Typegraph {
+            id: "https://metatype.dev/specs/0.0.3.json".into(),
+            path: None,
+            policies: vec![],
+            deps: vec![],
+            meta: TypeMeta {
+                ..Default::default()
+            },
+            runtimes: vec![common::typegraph::runtimes::TGRuntime::Unknown(
+                common::typegraph::runtimes::UnknownRuntime {
+                    name: "wasm".into(),
+                    data: Default::default(),
+                },
+            )],
+            materializers: vec![Materializer {
+                name: "function".into(),
+                runtime: 0,
```

Comment:
When I think of a concrete use case
```python
wasm = WasmRuntime() # WasmedgeRuntime renamed

g.expose(
# ..
add = wasm.fromMdk(
   t.struct({ a.., b.. }),
   t.integer(),
   wasm.fromMdk("custom.wasm") 
)
# ..
)
```
Then I believe the materializer name should be autogenerated (`custom_ ${someHashOrId}` maybe?) since on typegate we have to identify which one should be considered as custom from the materializer name.

Reply:
This has to happen on the sdk (probing inside `custom.wasm` to get some info could be difficult to do inside the sdk), but we can use the filename I guess.

Reply:
Won't filename get in the way when you're defining multiple materializer functions inside a single wasm module?

Reply:
> sdk

We ought to avoid reading any information from the wasm module at in the SDK. It might prove tricky for generation since this would imply a cyclical dependency.  That is:
- Wasm module depends on the codegen
- Codegen depends on the typegrpah/sdk
- Sdk depends on the wasm module

Reply:
Why was it you wanted to avoid `op_name` again?